plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoConnectorTest.java

```diff
@@ -15,12 +15,15 @@
 
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
```

Comment:
There's still correctness issue. 
Do we have a test case for this scenario? Sorry if I am missing. 
```sql
create table test2 (c1 row(a int, "b.c" int));
insert into test2 select row(1, 2);
select * from test2;
      c1
--------------
 {a=1, b.c=2}

set session mongodb.projection_pushdown_enabled = false;
select c1."b.c" from test2;
 b.c
-----
   2

set session mongodb.projection_pushdown_enabled = true;
select c1."b.c" from test2;
 b.c
------
 NULL
```

Reply:
Additionally can we add test coverage,  field with "$" as a prefix like 
`create table test2 (c1 row("$a" int, "b.c" int));`

Reply:
Added test case as a preparatory commit `Add test for dotted field and dollar prefixed field in mongo connector`.

These kinds of fields with dot(.) notation and dollar($) prefixed do not get pushdown actually in the current implementation. I will follow up with a PR for this.